src/test/jtx/envconfig.h

```diff
@@ -0,0 +1,105 @@
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_TEST_JTX_ENVCONFIG_H_INCLUDED
+#define RIPPLE_TEST_JTX_ENVCONFIG_H_INCLUDED
+
+#include <ripple/core/Config.h>
+
+namespace ripple {
+namespace test {
+
+/// @brief initializes a config object for use with jtx::Env
+///
+/// @param config the configuration object to be initialized
+extern
+void
+setupConfigForUnitTests (Config& config);
+
+namespace jtx {
+
+/// @brief creates and initializes a default
+/// configuration for jtx::Env
+///
+/// @return unique_ptr to Config instance
+inline
+std::unique_ptr<Config>
+envconfig()
+{
+    auto p = std::make_unique<Config>();
+    setupConfigForUnitTests(*p);
+    return p;
+}
+
+/// @brief creates and initializes a default configuration for jtx::Env and
+/// invokes the provided function/lambda with the configuration object.
+///
+/// @param modfunc callable function or lambda to modify the default config.
+/// The first argument to the function must be ripple::Config &.
+///
+/// @param args additional arguments that will be passed to
+/// the config modifier function (optional)
+///
+/// @return unique_ptr to Config instance
+template <class F, class... Args>
+std::unique_ptr<Config>
+envconfig(F&& modfunc, Args&&... args)
+{
+    auto p = envconfig();
+    modfunc(*p, std::forward<Args>(args)...);
+    return p;
+}
+
+/// @brief adjust config so no admin ports are enabled
+///
+/// this is intended for use with envconfig, as in
+/// envconfig(no_admin)
+///
+/// @param Config config instance to be modified
+void
+no_admin(Config&);
+
+/// @brief adjust configuration with params needed to be a validator
+///
+/// this is intended for use with envconfig, as in
+/// envconfig(validator, myseed)
+///
+/// @param cfg config instance to be modified
+/// @param seed seed string for use in secret key generation. A fixed default
+/// value will be used if this string is empty
+void
+validator(Config&, std::string const&);
+
+/// @brief adjust the default configured server ports by a specified value
+///
+/// This is intended for use with envconfig, as in
+/// envconfig(portincrement, 5)
+///
+/// @param cfg config instance to be modified
+/// @param int amount by which to increment the existing server port
+/// values in the config
+void
+portincrement(Config&, int);
```

Comment:
Nit: The name `portincrement` is neither camelCase nor snake_case. I know jtx doesn't always conform to the rippled style guild, but I think this name is out of bounds (same with the other multiple word names that run together).